src/libraries/System.Net.Sockets/tests/FunctionalTests/SendFile.cs

```diff
@@ -328,9 +326,10 @@ public async Task SendFileGetsCanceledByDispose()
                     }
                     else
                     {
-                        Assert.True(thrownDisposed);
+                        // TODO: how to make the test pass?
+                        //Assert.True(thrownDisposed);
```

Comment:
Cancellation is not implemented, so these test failed when `UsesSync = false`.
How to handle this correctly until we have cancellation?

Reply:
Even though cancellation isn't supported yet, this test should still pass. That is, Dispose should cancel outstanding operations.

What behavior are you seeing here?

Reply:
Is it just a matter of the wrong exception? If so that may be a test issue... we should look at the similar test for Send and make sure we are testing the same thing here.

Reply:
Here's the relevant code for Send and Receive: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs#L1020

I think we need to update the test here to match the logic there.

Reply:
> Is it just a matter of the wrong exception?

That's it, now it looks good. Thanks!